serial_communication/onboard/onboard/main.cpp

```diff
@@ -6,15 +6,93 @@
 #include "commDataFromTeensy.hpp" // header file for data from teensy to dispatcher
 #include "Joint.h" // struct that stores joint data
 #include "ConstJoint.h" // struct for constant joint data
+#include "ROM_DATA.h" // array to data addresses and enum
 
 LCMSerialSlave lcm; //initialize LCM object
 
+//PID Variables
+uint16_t minPot;
+uint16_t maxPot;
+
+//State Machine Variables
+volatile unsigned long int timer = 0;
+volatile bool allowPD = true; //flag for allowing the control
+
+
+//takes number of link/joint of teensy and an attribute
+// attributes: 0 - zeroTheta, 1 - minPot, 2 - maxPot, 3 - orientation
+int readROM(int link, int attribute) {
+  int address1 = ROM[(link*2-1) + attribute*6 + 2];
+  int address2 = ROM[(link*2) + attribute*6 + 2];
+  int val = EEPROM.read(address1);
+  val = val + (EEPROM.read(address2) << 8);
+  return val;
+}
+
+void setOrientROM(ConstJoint* cjoint){
+  EEPROM.write(ROM[20+cjoint->link*2-1], (byte)(cjoint->direction));
+  EEPROM.write(ROM[20+cjoint->link*2], (byte)0);
```

Comment:
As above, if you're doing strange index arithmetic, be clear about what's happening.